src/Lykke.ClientGenerator/Caching/CachingCallsWrapper.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading.Tasks;
+using Lykke.ClientGenerator.Infrastructure;
+using Microsoft.Extensions.Caching.Memory;
+using Newtonsoft.Json;
+using Polly;
+using Polly.Caching;
+using Polly.Caching.MemoryCache;
+using Refit;
+
+namespace Lykke.ClientGenerator.Caching
+{
+    /// <summary>
+    /// Abstract class to add caching to method calls.
+    /// Derive and implement <see cref="GetCachingTime"/>() to implement custom caching logic. 
+    /// </summary>
+    public abstract class CachingCallsWrapper : ICallsWrapper
```

Comment:
It looks ok for now, but in common case, it's better to avoid functionality extension by inheritance. Easy to implement sub class incorrectly and it lead to more tigh coupling. Consider to introduce something like ICachingTimeProvider (or to be more generic ICachingMetamodelProvider) as extension point abstraction to retrieve caching time.